client/app/services/designer-state.service.js

```diff
@@ -141,12 +105,19 @@
 
       function matchAtomicServiceItemToTabs(srvTemplate) {
         var subTab;
-        var newItem;
         if (srvTemplate.service_type === 'atomic') {
           if (srvTemplate.prov_type !== 'generic') {
             subTab = findSubTabByProvType(srvTemplate.prov_type);
-          } else {
+            if (!subTab) {
+              console.log("Couldn't find toolbox Tab for Prov_Type: " + srvTemplate.prov_type);
```

Comment:
Hi @AllenBW , I agree, but I'd like to fix that everywhere in a separate PR.  I'm currently working on changing the GUI to set/get mostly all values from 'ui_properties', a big PR.  I'm hoping to have some time at the end of this sprint to do some much needed eslint angular fixes.  -thanks!